float32 distance
bool distance_provided
float32 angle
bool angle_provided